### PR TITLE
Update AddressBook to ReadyBakey

### PR DESCRIPTION
**Pull request by punnyhuimin**
Updating of all occurrences of AddressBook. Please help me do a second pass to see if there are still any occurrences of it. Noticed there are some Javadoc comments that may or may not contain address book. Hoping that I managed to catch all of them.

Changed the `addressbook.json` to be `readybakey.json` instead. You can delete the `addressbook.json` and it will generate the new `readybakey.json` in the data folder.

Noticed that the AddRemark.md is very outdated. Let's try to see how we can further improve this and keep it updated.

Part of #168

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -48,16 +45,17 @@
...
 
         AppParameters appParameters = AppParameters.parse(getParameters());
         config = initConfig(appParameters.getConfigPath());
 
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         UserPrefs userPrefs = initPrefs(userPrefsStorage);
-        AddressBookStorage addressBookStorage = new JsonAddressBookStorage(userPrefs.getAddressBookFilePath());
-        storage = new StorageManager(addressBookStorage, userPrefsStorage);
+        seedu.address.storage.ReadyBakeyStorage readyBakeyStorage =
+                new JsonReadyBakeyStorage(userPrefs.getAddressBookFilePath());
```
Comment (darshan152) [inline src/main/java/seedu/address/MainApp.java:57]: Do we want to change the method name? `getAddressBookFilePath()`

File: src/main/java/seedu/address/model/UserPrefs.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public UserPrefs(ReadOnlyUserPrefs userPrefs) {
     public void resetData(ReadOnlyUserPrefs newUserPrefs) {
         requireNonNull(newUserPrefs);
         setGuiSettings(newUserPrefs.getGuiSettings());
-        setAddressBookFilePath(newUserPrefs.getAddressBookFilePath());
+        setReadyBakeyFilePath(newUserPrefs.getAddressBookFilePath());
```
Comment (darshan152) [inline src/main/java/seedu/address/model/UserPrefs.java:38]: Same method name as mentioned

File: src/main/java/seedu/address/model/UserPrefs.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public Path getAddressBookFilePath() {
         return addressBookFilePath;
     }
 
-    public void setAddressBookFilePath(Path addressBookFilePath) {
+    public void setReadyBakeyFilePath(Path addressBookFilePath) {
```
Comment (darshan152) [inline src/main/java/seedu/address/model/UserPrefs.java:54]: Change var name?

File: src/main/java/seedu/address/model/UserPrefs.java [hunk 4]
```diff
@@ -51,7 +51,7 @@ public Path getAddressBookFilePath() {
...
         requireNonNull(addressBookFilePath);
         this.addressBookFilePath = addressBookFilePath;
```
Comment (darshan152) [inline src/main/java/seedu/address/model/UserPrefs.java:56]: Change attribute name?

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -61,8 +59,8 @@
         };
     }
 
-    public static ReadOnlyAddressBook getSampleAddressBook() {
-        AddressBook sampleAb = new AddressBook();
+    public static seedu.address.model.ReadOnlyReadyBakey getSampleAddressBook() {
```
Comment (darshan152) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:62]: change Method name?

File: src/main/java/seedu/address/storage/ReadyBakeyStorage.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    Optional<seedu.address.model.ReadOnlyReadyBakey> readReadyBakey(Path filePath) throws DataConversionException,
+            IOException;
+
+    /**
+     * Saves the given {@link ReadOnlyReadyBakey} to the storage.
+     * @param addressBook cannot be null.
+     * @throws IOException if there was any problem writing to the file.
+     */
+    void saveReadyBakey(seedu.address.model.ReadOnlyReadyBakey addressBook) throws IOException;
```
Comment (darshan152) [inline src/main/java/seedu/address/storage/ReadyBakeyStorage.java:39]: change name of var?

File: src/main/java/seedu/address/storage/ReadyBakeyStorage.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    /**
+     * @see #saveReadyBakey(ReadOnlyReadyBakey)
+     */
+    void saveReadyBakey(seedu.address.model.ReadOnlyReadyBakey addressBook, Path filePath) throws IOException;
```
Comment (darshan152) [inline src/main/java/seedu/address/storage/ReadyBakeyStorage.java:44]: Change of var?

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 8]
```diff
@@ -24,10 +24,8 @@
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.Model;
 import seedu.address.model.ModelManager;
-import seedu.address.model.ReadOnlyAddressBook;
 import seedu.address.model.UserPrefs;
 import seedu.address.model.person.Person;
-import seedu.address.storage.JsonAddressBookStorage;
```
Comment (darshan152) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:30]: I see a few import statements with AB in the name removed, but the new ones dont seem to be imported, not sure if it matters?

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 9]
```diff
@@ -70,9 +68,10 @@ public void execute_validCommand_success() throws Exception {
 
     @Test
     public void execute_storageThrowsIoException_throwsCommandException() {
-        // Setup LogicManager with JsonAddressBookIoExceptionThrowingStub
-        JsonAddressBookStorage addressBookStorage =
-                new JsonAddressBookIoExceptionThrowingStub(temporaryFolder.resolve("ioExceptionAddressBook.json"));
+        // Setup LogicManager with JsonReadyBakeyIoExceptionThrowingStub
+        seedu.address.storage.JsonReadyBakeyStorage addressBookStorage =
+                new seedu.address.logic.LogicManagerTest.JsonReadyBakeyIoExceptionThrowingStub(
+                        temporaryFolder.resolve("ioExceptionAddressBook.json"));
```
Comment (darshan152) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:74]: Addressbook.json here, not sure if should/can be changed

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 10]
```diff
@@ -70,9 +68,10 @@ public void execute_validCommand_success() throws Exception {
 
     @Test
     public void execute_storageThrowsIoException_throwsCommandException() {
-        // Setup LogicManager with JsonAddressBookIoExceptionThrowingStub
-        JsonAddressBookStorage addressBookStorage =
-                new JsonAddressBookIoExceptionThrowingStub(temporaryFolder.resolve("ioExceptionAddressBook.json"));
+        // Setup LogicManager with JsonReadyBakeyIoExceptionThrowingStub
+        seedu.address.storage.JsonReadyBakeyStorage addressBookStorage =
```
Comment (darshan152) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:72]: name of variable

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 11]
```diff
@@ -149,13 +148,14 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
     /**
      * A stub class to throw an {@code IOException} when the save method is called.
      */
-    private static class JsonAddressBookIoExceptionThrowingStub extends JsonAddressBookStorage {
-        private JsonAddressBookIoExceptionThrowingStub(Path filePath) {
+    private static class JsonReadyBakeyIoExceptionThrowingStub extends seedu.address.storage.JsonReadyBakeyStorage {
```
Comment (darshan152) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:151]: mentions the whole name, should be imported instead?

Comment (darshan152) [reply src/test/java/seedu/address/logic/LogicManagerTest.java:151]: happens a few times in other files

File: src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java [hunk 12]
```diff
@@ -73,7 +73,7 @@ public void readUserPrefs_extraValuesInFile_extraValuesIgnored() throws DataConv
     private UserPrefs getTypicalUserPrefs() {
         UserPrefs userPrefs = new UserPrefs();
         userPrefs.setGuiSettings(new GuiSettings(1000, 500, 300, 100));
-        userPrefs.setAddressBookFilePath(Paths.get("addressbook.json"));
+        userPrefs.setReadyBakeyFilePath(Paths.get("addressbook.json"));
```
Comment (darshan152) [inline src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java:76]: Did we change the path of this to readybakey.json?

**Review by darshan152 — COMMENTED** [review]
Few comments to have a look at

**Comment by punnyhuimin** [comment]
Keeping this aside. Not a priority and will set this as draft.

**Comment by punnyhuimin** [comment]
Closing due to unnecessary code changes.

**Pull request closed without merging**